src/internal/types.test.ts

```diff
@@ -0,0 +1,99 @@
+import { type Branded, type IfSimpleRecord } from "./types";
+
+describe("IfSimpleRecord", () => {
+  test("string, number, symbol", () => {
+    expectTypeOf<
+      IfSimpleRecord<Record<string, unknown>>
+    >().toEqualTypeOf<true>();
+
+    expectTypeOf<
+      IfSimpleRecord<Record<number, unknown>>
+    >().toEqualTypeOf<true>();
+
+    expectTypeOf<
+      IfSimpleRecord<Record<number | string, unknown>>
+    >().toEqualTypeOf<true>();
+
+    expectTypeOf<
+      IfSimpleRecord<Record<symbol, unknown>>
+    >().toEqualTypeOf<true>();
+  });
```

Comment:
no reason not to split them up, and maybe have a slightly more descriptive name 🙄 ?